File: GoLang/剑指Offer/easy/剑指Offer_21_调整数组顺序使奇数位于偶数前面.go
package easy

func exchange(nums []int) []int {
	left, right := 0, len(nums)-1
	for left < right {
		for left < right && nums[left]&1 == 1 {
			left++
		}
		for left < right && nums[right]&1 == 0 {
			right--
		}
		temp := nums[left]
		nums[left] = nums[right]
		nums[right] = temp
	}
	return nums
}

File: Java/剑指Offer/easy/剑指Offer_21_调整数组顺序使奇数位于偶数前面.java
package easy;

/**
 * Created by lwd at 2021/7/13
 *
 * @Description:
 */
public class 剑指Offer_21_调整数组顺序使奇数位于偶数前面 {
    public int[] exchange(int[] nums) {
        if (nums == null || nums.length <= 1) return nums;

        int left = 0, right = nums.length - 1;
        while (left < right) {
            while ((nums[left] & 1) == 1 && (left < right)) {
                left++;
            }
            while ((nums[right] & 1) == 0 && (left < right)) {
                right--;
            }
            int temp = nums[left];
            nums[left] = nums[right];
            nums[right] = temp;
        }
        return nums;
    }
}
